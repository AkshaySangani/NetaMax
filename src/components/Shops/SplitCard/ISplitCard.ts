import { IconProps } from "@chakra-ui/icon";

export interface ISplitCardProps {
  /**
   * The title1 for SplitCard
   * @type {String}
   *
   * */
  firstTitle: string;
  /**
   * The subtitle1 for SplitCard
   * @type {String}
   *
   * */
  firstSubtitle?: string;
  /**
   * The icon1 for SplitCard
   * @type {IconProps}
   *
   * */
  firstIcon: IconProps;
  /**
   * The title1 for SplitCard
   * @type {String}
   *
   * */
  secondTitle: string;
  /**
   * The subtitle2 for SplitCard
   * @type {String}
   *
   * */
  secondSubtitle?: string | number;
  /**
   * The icon2 for SplitCard
   * @type {IconProps}
   *
   * */
  secondIcon: IconProps;
  /**
   * The title3 for SplitCard
   * @type {String}
   *
   * */
  thirdTitle: string;
  /**
   * The subtitle3 for SplitCard
   * @type {String}
   *
   * */
  thirdSubtitle: string;
  /**
   * The icon3 for SplitCard
   * @type {IconProps}
   *
   * */
  thirdIcon: IconProps;
}
