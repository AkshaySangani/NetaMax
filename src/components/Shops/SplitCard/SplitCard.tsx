import React, { ReactElement } from "react";

import { Center, Divider, Flex, Text } from "@chakra-ui/layout";
import { Box } from "@chakra-ui/react";

import { ISplitCardProps } from "./ISplitCard";

/**
 * Card divided into 3 sections component.
 * @returns {ReactElement} the SplitCard component.
 * @param {ISplitCardProps} props SplitCard props
 */
export const SplitCard = (props: ISplitCardProps): ReactElement => {
  const {
    firstTitle,
    firstSubtitle,
    firstIcon,
    secondTitle,
    secondSubtitle,
    secondIcon,
    thirdTitle,
    thirdSubtitle,
    thirdIcon,
  } = props;

  return (
    <Flex
      direction={["column", "column", "row", "row"]}
      mt="6"
      mb="6"
      boxShadow="xs"
      p={["1", "2", "4"]}
      rounded="lg"
      justify="center"
    >
      <Flex align="center" mx="2" mt={["2", "0"]}>
        {firstIcon}
        <Box ml="2" mx="4" width={["auto", "auto", "130px"]}>
          <Text fontSize={{ base: "17px", md: "14px", lg: "15px" }}>
            {firstTitle}
            <b> {firstSubtitle}</b>
          </Text>
        </Box>
      </Flex>
      <Center
        align="center"
        height={["0px", "0px", "40px"]}
        ml={["0", "0", "3", "8"]}
        mr={["0", "0", "3", "8"]}
      >
        <Divider orientation="vertical" />
      </Center>
      <Flex align="center" mx="2" mt={["2", "0"]}>
        {secondIcon}
        <Box ml="2" mx="4" width={["auto", "auto", "100px"]}>
          <Text fontSize={{ base: "17px", md: "14px", lg: "15px" }}>
            {secondTitle}
            <b>{secondSubtitle}</b>
          </Text>
        </Box>
      </Flex>
      <Center
        align="center"
        height={["0px", "0px", "40px"]}
        ml={["0", "0", "3", "8"]}
        mr={["0", "0", "3", "8"]}
      >
        <Divider orientation="vertical" />
      </Center>
      <Flex align="center" mx="2" mt={["2", "0"]} mb={["2", "0"]}>
        {thirdIcon}
        <Box ml="2" mx="4" width={["auto", "auto", "140px"]}>
          <Text fontSize={{ base: "17px", md: "14px", lg: "15px" }}>
            {thirdTitle}
            <b>{thirdSubtitle}</b>
          </Text>
        </Box>
      </Flex>
    </Flex>
  );
};
