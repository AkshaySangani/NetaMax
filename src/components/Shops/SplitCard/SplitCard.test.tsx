import { render, cleanup } from "@testing-library/react";
import { SplitCard } from "./SplitCard";
import { ISplitCardProps } from "./ISplitCard";
import IconDetailOrder from "styled/icons/Shops/Order/OrderNumber";
import IconDetailTelephone from "styled/icons/Shops/Order/Telephone";
import IconDetailUser from "styled/icons/Shops/Order/User";

describe("SplitCard Component", () => {
  let mock: ISplitCardProps;

  beforeEach(() => {
    mock = {
      firstTitle: "Nombre cliente: ",
      firstSubtitle: "Order 1",
      secondTitle: "Teléfono: ",
      secondSubtitle: "123456789",
      thirdTitle: "Número de orden: ",
      thirdSubtitle: "ANMX123",
      firstIcon: <IconDetailUser />,
      secondIcon: <IconDetailTelephone />,
      thirdIcon: <IconDetailOrder />,
    };
  });

  afterEach(() => {
    cleanup();
  });

  test("SplitCard should be render correctly", () => {
    const { container } = render(<SplitCard {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
