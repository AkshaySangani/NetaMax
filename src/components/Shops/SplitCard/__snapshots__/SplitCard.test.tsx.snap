// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplitCard Component SplitCard should be render correctly 1`] = `
<div>
  <div
    class="css-1itlcov"
  >
    <div
      class="css-2gfq85"
    >
      <svg
        class="chakra-icon css-onkibi"
        focusable="false"
        viewBox="0 0 45 45"
      >
        <circle
          cx="22.5"
          cy="22.5"
          fill="#D6E1FF"
          r="22.5"
        />
        <path
          d="M26.7776 24.2383C26.6718 24.1926 26.5796 24.1206 26.5098 24.0292C26.4399 23.9378 26.3948 23.8301 26.3788 23.7163C26.3627 23.6025 26.3762 23.4865 26.418 23.3794C26.4598 23.2723 26.5285 23.1777 26.6175 23.1047C27.7313 22.2638 28.556 21.0992 28.9779 19.7715C29.3999 18.4438 29.3983 17.0183 28.9734 15.6915C28.6426 14.6316 28.0507 13.6713 27.2518 12.8981C26.4529 12.1249 25.4725 11.5635 24.3999 11.2652C23.3888 10.9747 22.3239 10.922 21.2888 11.1113C20.2538 11.3006 19.277 11.7267 18.4352 12.356C17.5933 12.9854 16.9095 13.8009 16.4374 14.7383C15.9653 15.6758 15.7178 16.7096 15.7145 17.7585C15.7155 18.7916 15.9561 19.8105 16.4175 20.7356C16.879 21.6606 17.5487 22.4668 18.3744 23.0912C18.4651 23.1629 18.5356 23.2568 18.5791 23.3638C18.6225 23.4707 18.6375 23.5871 18.6225 23.7015C18.6074 23.8159 18.5629 23.9245 18.4933 24.0167C18.4237 24.1089 18.3313 24.1815 18.2251 24.2275C16.6639 25.0068 15.3495 26.2016 14.4274 27.6795C13.5052 29.1574 13.0113 30.8608 13.0003 32.6012C12.9967 32.782 13.029 32.9618 13.0953 33.1301C13.1617 33.2984 13.2608 33.452 13.3869 33.582C13.5131 33.712 13.6637 33.8159 13.8303 33.8876C13.9968 33.9593 14.176 33.9975 14.3574 34H30.6426C30.824 33.9975 31.0032 33.9593 31.1697 33.8876C31.3363 33.8159 31.4869 33.712 31.6131 33.582C31.7392 33.452 31.8383 33.2984 31.9047 33.1301C31.971 32.9618 32.0033 32.782 31.9997 32.6012C31.9873 30.8629 31.4929 29.1618 30.5714 27.686C29.6498 26.2101 28.3368 25.017 26.7776 24.2383Z"
          fill="#0035BD"
        />
      </svg>
      <div
        class="css-hdej0o"
      >
        <p
          class="chakra-text css-1caq11m"
        >
          Nombre cliente: 
          <b>
             
            Order 1
          </b>
        </p>
      </div>
    </div>
    <div
      align="center"
      class="css-14z2clc"
    >
      <hr
        aria-orientation="vertical"
        class="chakra-divider css-12gj880"
      />
    </div>
    <div
      class="css-2gfq85"
    >
      <svg
        class="chakra-icon css-onkibi"
        focusable="false"
        viewBox="0 0 45 45"
      >
        <circle
          cx="22.5"
          cy="22.5"
          fill="#D6E1FF"
          r="22.5"
        />
        <path
          d="M28.1676 26.162C27.9355 25.9298 27.6599 25.7456 27.3565 25.6199C27.0532 25.4942 26.728 25.4295 26.3996 25.4295C26.0713 25.4295 25.7461 25.4942 25.4428 25.6199C25.1394 25.7456 24.8638 25.9298 24.6316 26.162L24.2466 26.547C22.1519 24.7894 20.2149 22.852 18.4576 20.757L18.8436 20.372C19.3123 19.9032 19.5756 19.2674 19.5756 18.6045C19.5756 17.9416 19.3123 17.3058 18.8436 16.837L16.7216 14.716C16.4895 14.4836 16.2138 14.2993 15.9103 14.1735C15.6069 14.0478 15.2816 13.983 14.9531 13.983C14.6247 13.983 14.2994 14.0478 13.996 14.1735C13.6925 14.2993 13.4168 14.4836 13.1846 14.716L12.0216 15.879C11.4548 16.4507 11.1017 17.1998 11.0214 18.0008C10.9411 18.8019 11.1386 19.6062 11.5806 20.279C15.0402 25.4937 19.5082 29.9635 24.7216 33.425C25.3964 33.8616 26.1997 34.0556 26.9993 33.9751C27.799 33.8946 28.5475 33.5443 29.1216 32.982L30.2866 31.817C30.755 31.3482 31.0181 30.7127 31.0181 30.05C31.0181 29.3873 30.755 28.7518 30.2866 28.283L28.1676 26.162Z"
          fill="#0035BD"
        />
        <path
          d="M34.0002 10.007H29.2002C29.0031 10.0084 28.8108 10.0681 28.6474 10.1785C28.4841 10.2889 28.357 10.4451 28.2822 10.6275C28.2074 10.81 28.1882 11.0104 28.227 11.2037C28.2658 11.397 28.3608 11.5746 28.5002 11.714L29.8342 13.05L23.9342 18.944C23.653 19.2254 23.495 19.607 23.4951 20.0048C23.4952 20.4027 23.6533 20.7842 23.9347 21.0655C24.2161 21.3468 24.5977 21.5047 24.9956 21.5046C25.3935 21.5045 25.775 21.3464 26.0562 21.065L31.9562 15.171L33.2902 16.506C33.3834 16.5994 33.494 16.6735 33.6159 16.7239C33.7377 16.7744 33.8684 16.8002 34.0002 16.8C34.1317 16.7995 34.2618 16.7733 34.3832 16.723C34.5658 16.6473 34.7218 16.5193 34.8316 16.355C34.9414 16.1907 35.0001 15.9976 35.0002 15.8V11.007C35.0002 10.7418 34.8949 10.4874 34.7074 10.2999C34.5198 10.1123 34.2655 10.007 34.0002 10.007Z"
          fill="#0035BD"
        />
      </svg>
      <div
        class="css-jf7yai"
      >
        <p
          class="chakra-text css-1caq11m"
        >
          Teléfono: 
          <b>
            123456789
          </b>
        </p>
      </div>
    </div>
    <div
      align="center"
      class="css-14z2clc"
    >
      <hr
        aria-orientation="vertical"
        class="chakra-divider css-12gj880"
      />
    </div>
    <div
      class="css-l828e9"
    >
      <svg
        class="chakra-icon css-onkibi"
        focusable="false"
        viewBox="0 0 45 45"
      >
        <circle
          cx="22.5"
          cy="22.5"
          fill="#D6E1FF"
          r="22.5"
        />
        <path
          d="M31.7673 11.7333H18.4339C17.0554 11.7333 15.9339 12.8548 15.9339 14.2333V15.0667H15.1006C13.7221 15.0667 12.6006 16.1881 12.6006 17.5667V30.9C12.6006 32.2785 13.7221 33.4 15.1006 33.4H28.4339C29.8124 33.4 30.9339 32.2785 30.9339 30.9V30.0667H31.7673C33.1458 30.0667 34.2673 28.9452 34.2673 27.5667V14.2333C34.2673 12.8548 33.1458 11.7333 31.7673 11.7333ZM27.6006 29.65C27.6006 29.8801 27.414 30.0667 27.1839 30.0667H16.3506C16.1205 30.0667 15.9339 29.8801 15.9339 29.65V28.8167C15.9339 28.5866 16.1205 28.4 16.3506 28.4H27.1839C27.414 28.4 27.6006 28.5866 27.6006 28.8167V29.65ZM27.6006 26.3167C27.6006 26.5468 27.414 26.7333 27.1839 26.7333H16.3506C16.1205 26.7333 15.9339 26.5468 15.9339 26.3167V25.4833C15.9339 25.2532 16.1205 25.0667 16.3506 25.0667H27.1839C27.414 25.0667 27.6006 25.2532 27.6006 25.4833V26.3167ZM27.6006 22.9833C27.6006 23.2134 27.414 23.4 27.1839 23.4H16.3506C16.1205 23.4 15.9339 23.2134 15.9339 22.9833V22.15C15.9339 21.9199 16.1205 21.7333 16.3506 21.7333H27.1839C27.414 21.7333 27.6006 21.9199 27.6006 22.15V22.9833ZM27.6006 19.65C27.6006 19.8801 27.414 20.0667 27.1839 20.0667H16.3506C16.1205 20.0667 15.9339 19.8801 15.9339 19.65V18.8167C15.9339 18.5866 16.1205 18.4 16.3506 18.4H27.1839C27.414 18.4 27.6006 18.5866 27.6006 18.8167V19.65ZM32.6006 27.5667C32.6006 28.0261 32.2267 28.4 31.7673 28.4H30.9339V17.5667C30.9339 16.1881 29.8124 15.0667 28.4339 15.0667H17.6006V14.2333C17.6006 13.7739 17.9745 13.4 18.4339 13.4H31.7673C32.2267 13.4 32.6006 13.7739 32.6006 14.2333V27.5667Z"
          fill="#0035BD"
        />
      </svg>
      <div
        class="css-84b9i6"
      >
        <p
          class="chakra-text css-1caq11m"
        >
          Número de orden: 
          <b>
            ANMX123
          </b>
        </p>
      </div>
    </div>
  </div>
</div>
`;
